### ISSUE TITLE
pgp select fails if secret stored on card

### ISSUE THREAD

**Issue by OJFord** [keybase/keybase-issues#3378]
I have a Yubikey, and the public key (only) on a Keybase-verified computer's GPG keychain.

`gpg --list-keys` shows the key, `gpg --card-status` shows the secret's available, I can use the sub keys no problem with git, ssh, etc., but `keybase pgp select` says:
> ▶ ERROR No PGP keys available to choose from.

Is there a way of manually signing the public key, and adding it to Keybase's chain?

**Comment by maxtaco** [comment]
This used to work, I'm surprised it doesn't.  Can you run `keybase log send` and we can take a look at what went wrong? Thanks....

**Comment by OJFord** [comment]
Thanks - 8c5b9c552e20dee995487f1c

**Comment by OJFord** [comment]
@zapu helped me resolve this in chat.

Turned out it was because I have my `GNUPGHOME` set to non-default location (not `~/.gnupg`) and this setting is not being passed through to the gpg command Keybase.app runs - or at least not if it wasn't in the environment from which Keybase.app started.

Solution/workaround was to kill Keybase.app and run `keybase --standalone pgp select` from a shell with the correct environment.

I'll close this issue since I've sorted my problem, and I assume Keybase has separate internal tracking if there's any desire to improve behaviour or the way it fails.

Many thanks again Zapu, it was a refreshing treat to be able to chat to you directly and get to the bottom of the issue pretty quickly 🙂.
